aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -513,18 +513,39 @@ const Tensor& resize_sparse_csr_(
   return self;
 }
 
-Tensor& copy_sparse_csr_(Tensor& self, const Tensor& src, bool non_blocking) {
+Tensor& copy_sparse_compressed_(Tensor& self, const Tensor& src, bool non_blocking) {
+  AT_DISPATCH_ALL_SPARSE_COMPRESSED_LAYOUTS(self.layout(), "copy_sparse_compressed_", [&]{});
   TORCH_CHECK(
-      self.is_sparse_csr() && src.is_sparse_csr(),
-      "copy_sparse_csr_: copy between different layouts is not supported. Found self type = ",
-      self.toString(),
-      " and src type = ",
-      src.toString());
+      self.layout() == src.layout(),
+      "torch.copy_: copy of sparse compressed tensors having different layouts is not supported.",
+      " self layout is ", self.layout(), " and src layout is ", src.layout());
   TORCH_CHECK(
-      self._nnz() == src._nnz(),
-      "copy_sparse_csr_: only tensors with the same number of specified elements are supported.");
-  self.crow_indices().copy_(src.crow_indices(), non_blocking);
-  self.col_indices().copy_(src.col_indices(), non_blocking);
+      self._nnz() == src._nnz(),  // actually, values copy allows different shapes as long as operands are broadcastable
+      "torch.copy_: only sparse compressed tensors with the same number of specified elements are supported.");
+  auto self_compressed_dims = self.size(compressedDimension(self.layout(), self.sizes()));
+  auto src_compressed_dims = src.size(compressedDimension(src.layout(), src.sizes()));
+  TORCH_CHECK(self_compressed_dims == src_compressed_dims,
```

Comment:
I'd state the dimension index as well, since we don't really have dedicated APIs to Python users for compressed and plain indices etc. I'd say something like "Expected shape X and shape Y to match along dimension Z for given layouts A and B".

Reply:
Fixed.